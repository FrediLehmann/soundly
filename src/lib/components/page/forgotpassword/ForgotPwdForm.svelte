<script lang="ts" context="module">
  import { Button, Input } from '$lib/components/atoms';

  const forgotPwdForm = 'ForgotPwd';
</script>

<script lang="ts">
  export let resetPassword: (arg0: string) => void;
  export let submitting: boolean;

  let emailError = '';

  const submit = (e: any) => {
    resetPassword(e.target.email.value);
  };

  const click = async () => {
    const email = document.forms[forgotPwdForm].email;

    if (email.validity.valid) {
      document.getElementById(`${forgotPwdForm}_submit`).click();
      return;
    }

    emailError = email.validationMessage;
  };
</script>

<form on:submit|preventDefault={submit} name={forgotPwdForm}>
  <Input
    required
    type="email"
    name="email"
    label="Email"
    title="Please enter your email."
    error={emailError}
    disabled={submitting} />
  <Button disabled={submitting} on:click={click} style="primary" class="mbs-2">
    Reset password
  </Button>
  <input id={`${forgotPwdForm}_submit`} type="submit" hidden />
</form>

<style>
  form {
    display: flex;
    flex-direction: column;
    gap: 1rem;
  }
</style>
