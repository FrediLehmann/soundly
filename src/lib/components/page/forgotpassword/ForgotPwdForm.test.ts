/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import ForgotPwdForm from './ForgotPwdForm.svelte';

describe('<ForgotPwdForm /> component', () => {
  test('renders correctly', () => {
    const { component } = render(ForgotPwdForm, {
      props: { resetPassword: (_: unknown) => _, submitting: false }
    });
    expect(component).toMatchSnapshot();
  });
});
