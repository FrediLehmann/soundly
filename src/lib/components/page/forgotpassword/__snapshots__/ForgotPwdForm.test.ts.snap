// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPwdForm /> component renders correctly 1`] = `
ForgotPwdForm {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      false,
      "",
      [Function],
      [Function],
      [Function],
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "resetPassword": 4,
      "submitting": 0,
    },
    "root": <div>
      <form
        class="svelte-1i4vv2l"
        name="ForgotPwd"
      >
        <div
          class="container svelte-1ulq3bj"
          data-required="true"
          data-sr-only="false"
        >
          <label
            class="svelte-1ulq3bj"
            for="email"
          >
            Email
          </label>
           
          <input
            class="svelte-1ulq3bj"
            id="email"
            required=""
            title="Please enter your email."
            type="email"
          />
           
        </div>
         
        <button
          class="mbs-2 svelte-19wuz13"
          data-type="primary"
          type="button"
        >
          Reset password
        </button>
        
         
        <input
          hidden=""
          id="ForgotPwd_submit"
          type="submit"
        />
      </form>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
